src/pages/task/TestsTableCore.tsx

```diff
@@ -0,0 +1,248 @@
+  taskTests: [TaskTestsData];
+}
+
+const SpinWrapper = styled.div({
+  textAlign: "center",
+  paddingTop: 40,
+  paddingBottom: 40,
+  border: "1px solid #e8e8e8"
+});
+const loadMoreContent = (
+  <SpinWrapper>
+    <Spin tip="Loading..." />
+  </SpinWrapper>
+);
+
+const columns: Array<ColumnProps<TaskTestsData>> = [
+  {
+    title: "Name",
+    dataIndex: "testFile",
+    key: Categories.TestName,
+    sorter: true
+  },
+  {
+    title: "Status",
+    dataIndex: "status",
+    key: Categories.Status,
+    sorter: true,
+    render: (tag: string) => {
+      let color: string;
+      switch (tag) {
+        case TestStatus.Succeeded:
+          color = "green";
+          break;
+        case TestStatus.Failed:
+          color = "red";
+          break;
+        case TestStatus.SilentlyFailed:
+          color = "volcano";
+          break;
+        case TestStatus.Skipped:
+          color = "orange";
+          break;
+      }
+
+      return (
+        <span>
+          <Tag color={color} key={tag}>
+            {tag.toUpperCase()}
+          </Tag>
+        </span>
+      );
+    }
+  },
+  {
+    title: "Time",
+    dataIndex: "duration",
+    key: Categories.Duration,
+    sorter: true,
+    render: (text: number): string => {
+      return msToDuration(Math.trunc(text));
+    }
+  }
+];
+const rowKey = ({ id }) => id;
+type Props = ValidInitialQueryParams & Limit;
+export const TestsTableCore: React.FC<Props> = ({
+  initialSort,
+  initialCategory,
+  limit
+}) => {
+  const { taskID } = useParams();
+  const { search, pathname } = useLocation();
+  const { replace } = useHistory();
+  const { data, fetchMore, networkStatus } = useQuery<
+    TaskTestsData,
+    TakskTestsVars
+  >(TESTS_QUERY, {
+    variables: {
+      id: taskID,
+      dir: initialSort === Sort.Asc ? "ASC" : "DESC",
+      cat: initialCategory as Category,
+      pageNum: 0,
+      limitNum: limit
+    },
+    notifyOnNetworkStatusChange: true
+  });
+  const parsed = queryString.parse(search);
+  const category = (parsed[RequiredQueryParams.Category] || "")
+    .toString()
+    .toUpperCase();
+  const sort = parsed[RequiredQueryParams.Sort];
+  const prevCategory = usePrevious(category);
+  const prevSort = usePrevious(sort);
+  useEffect(() => {
+    const elements = document.querySelectorAll(
+      "th.ant-table-column-has-actions.ant-table-column-has-sorters"
+    );
+    if (networkStatus < NetworkStatus.ready) {
+      elements.forEach(el => {
+        (el as HTMLElement).style["pointer-events"] = "none";
+      });
+    } else {
+      elements.forEach(el => {
+        (el as HTMLElement).style["pointer-events"] = "auto";
+      });
+    }
+  }, [networkStatus]);
+
+  if (
+    (sort !== prevSort || category !== prevCategory) &&
+    networkStatus === NetworkStatus.ready
+  ) {
+    fetchMore({
+      variables: {
+        cat: category,
+        dir: sort === Sort.Asc ? "ASC" : "DESC",
+        pageNum: 0,
+        limitNum: limit
```

Comment:
it looks to me like `limit` does not need to be passed down through multiple levels of props because its just hardcoded when `<TestsTable limit={10} />` is rendered. Is there a reason that `limit` needs to be passed as a prop?

Reply:
I thought this is one of the values we are most likely to adjust so I made it available as prop TestsTable so it's easier to find and fine tune.

Reply:
i can't think of any reason that we would need to pass it down two levels of props. if it's being hardcoded and not passed in from state that changes then we don't need to pass the value down as props. you can leave `limit` as a prop and just give it a default value so that it does not need to be passed down in two places. 

Every prop that require to be passed to our components makes the api of the component more complex. therefore if there is no use case for this prop to be dynamic right now then we could remove it as a prop altogether and hardcode it in `TestTableCore`. Then when and if we need it we can expose it as a prop. Otherwise we just complicate the interface of our components for situations that may or may never exist.